lib/puppet/provider/service/systemd.rb

```diff
@@ -46,11 +67,20 @@ def status
   end
 
   def enable
+    output = systemctl("unmask", @resource[:name])
```

Comment:
My thought is always Puppet == Glactus of systems, destroyer of bad
practices.

Trevor

On Wed, Feb 11, 2015 at 3:40 PM, Kylo Ginsberg <notifications@github.com>
wrote:

> In lib/puppet/provider/service/systemd.rb
> <https://github.com/puppetlabs/puppet/pull/3141#discussion_r24531127>:
>
> > @@ -46,11 +67,20 @@ def status
> >    end
> >
> >    def enable
> > +    output = systemctl("unmask", @resource[:name])
>
> Yeah, I see the dilemma and I'm not sure either.
>
> I like the idea of enable respecting :mask, and I sorta think that part is
> non-controversial.
>
> But I'm not 100% sure what to do about the default for :mask. On the one
> hand, clearly the default for :mask should be false for standard uses (e.g.
> I didn't even know 'masked' was a thing until today). But if someone is
> using 'masked' to mean hands-off even for puppet ... This of course goes to
> your first bullet "puppet is authoritative".
>
> Thoughts?
>
> (Btw, I feel like I need to educate myself about usage of 'masked' to get
> a better feel for this whole topic.)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppet/pull/3141/files#r24531127>.
>



-- 
Trevor Vaughan
Vice President, Onyx Point, Inc
(410) 541-6699
tvaughan@onyxpoint.com

-- This account not approved for unencrypted proprietary information --